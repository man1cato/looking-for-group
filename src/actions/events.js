import axios from 'axios';
import moment from 'moment';
import _ from 'lodash';

const apiKey= 'keyzG8AODPdzdkhjG';
const baseUrl = 'https://api.airtable.com/v0/appOY7Pr6zpzhQs6l';

//GET_EVENTS
export const getEvents = (events) => ({
    type: 'GET_EVENTS',
    events
});

export const startGetEvents = (usersGroups, userAvailabilities) => {
    return async (dispatch) => {
        try {
            let events = [];
            for (let group of usersGroups) {                                    //FOR EACH OF THE USER'S GROUPS...
                const groupRecordId = group.id; 
                for (let availabilityRecordId of userAvailabilities) {                    
                    const filter = `AND({Group Record ID}="${groupRecordId}",{Availability Record ID}="${availabilityRecordId}")`;
                    const eventResponse = await axios.get(`${baseUrl}/Events?filterByFormula=${filter}&api_key=${apiKey}`); 
                    const event = eventResponse.data.records[0];
                    if (event) {
                        events.push({
                            id: event.id, 
                            interest: group.interest,
                            area: group.area,
                            availability: event.fields.Availability[0],
                            occurrence: event.fields.Occurrence,
                            startDatetime: event.fields['Start Date & Time'],
                            date: moment(event.fields['Start Date & Time']).format('MMM Do'),
                            startTime: moment(event.fields['Start Date & Time']).format('h:mm a')
                        });
                    }
                }
            }
            console.log('events:', events);
            const sortedEvents = _.orderBy(events, ['startDatetime'], ['asc']);
            dispatch(getEvents((sortedEvents)));
        } catch (e) {
            throw new Error('Failed to retrieve groups in startGetEvents');
        }
    };
};


// //UPDATE_EVENTS
// export const updateEvents = (groups) => ({
//     type: 'UPDATE_GROUPS',
//     groups
// });

// export const startUpdateGroups = (areaRecordId, userInterests) => {
//     return async (dispatch) => {
//         try {
//             let groups = [];
//             for (let interestRecordId of userInterests) {
//                 const filter = `AND({Interest Record ID}="${interestRecordId}",{Area Record ID}="${areaRecordId}")`;
//                 const response = await axios.get(`${baseUrl}/Groups?filterByFormula=${filter}&api_key=${apiKey}`);
//                 const group = response.data.records[0];
//                 groups.push({
//                     id: group.id, 
//                     area: group.fields["Area Text"],
//                     interest: group.fields["Interest Name"],
//                     availability: group.fields["Group Availibility"]
//                 });
//             }
//             const sortedGroups = _.orderBy(groups, ['interest'], ['asc']);
//             console.log('from startUpdateGroups:',sortedGroups);
//             dispatch(updateGroups(sortedGroups));
//         } catch (e) {
//             throw new Error('Update groups failed at startUpdateGroups');
//         }
        
//     };
// };