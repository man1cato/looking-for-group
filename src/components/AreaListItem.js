import React from 'react';
import {Link} from 'react-router-dom';

const AreaListItem = ({id, name}) => (
    <Link className="list-item" to={`/areas/${id}`}>                             
        <div>
            <h3 className="list-item__title">{name}</h3>
        </div>
    </Link>
);


export default AreaListItem;