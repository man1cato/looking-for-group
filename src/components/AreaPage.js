import React from 'react';
import {connect} from 'react-redux';
import getAreas from '../selectors/areas';

export const AreaPage = (props) => (
    <div>
        <h1>Areas</h1>
        <div>
            Put areas here
        </div>
    </div>
);

// const mapStateToProps = (state) => {
//     const view = 'Alpha Test List';
//     return {
//         areas: getAreas(view)
//     };
// };

// export default connect(mapStateToProps)(AreaPage);
