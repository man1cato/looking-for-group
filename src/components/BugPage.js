import React from 'react';

const BugPage = () => (
    <div>
        <iframe className="airtable-embed" src="https://airtable.com/embed/shrPbU1gbCEesGW0c?backgroundColor=orange" />
    </div>
);


export default BugPage;