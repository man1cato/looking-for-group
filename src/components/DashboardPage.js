import React from 'react';
import GroupList from './GroupList';


const DashboardPage = () => (
    <div>
        <div className="page-header">
            <div className="content-container">
                <h2 className="page-header__title">My Local Interest Groups</h2>
                <div>Groups will only appear for those interests that have a sufficient number of people in the area who share that interest.</div>
            </div>
        </div>
        <GroupList />
    </div>
);


export default DashboardPage;