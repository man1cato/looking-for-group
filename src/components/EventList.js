import React from 'react';
import {connect} from 'react-redux';
import EventListItem from './EventListItem';


export const EventList = (props) => (
    <div>
            {
                props.events.length === 0 ? (
                    <div className="list-item list-item--message">This may take some time to load. If it has been more than a minute, then there are no available event times for this group. </div>
                ) : (
                    props.events.map((event) => {
                        if (props.interest === event.interest) {
                            return (<EventListItem key={event.id} {...event} /> );
                        }
                    })
                )
            }
    </div>
);


const mapStateToProps = (state) => ({ events: state.events });

export default connect(mapStateToProps)(EventList);
