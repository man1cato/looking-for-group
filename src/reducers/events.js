//EVENTS REDUCER
const eventsReducer = (state = [], action) => {
    switch (action.type) {
        case 'GET_EVENTS':
            return action.events;
        case 'UPDATE_EVENTS':
            return action.events;
        default: 
            return state;
    }
};

export default eventsReducer;