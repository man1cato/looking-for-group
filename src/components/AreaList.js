import React from 'react';
import {connect} from 'react-redux';
import AreaListItem from './AreaListItem';


export const AreaList = (props) => (
    <div>
        <div className="page-header">
            <div className="content-container">
                <h2 className="page-header__title">Areas</h2>
                <span>Select the area in which you live.</span>
            </div>
        </div>
        <div className="content-container">
            <div className="list-body">
                {
                    props.areas.length === 0 ? (
                        <div className="list-item list-item--message">No areas</div>
                    ) : (
                        props.areas.map((area) => <AreaListItem key={area.id} {...area} /> )    
                    )
                }
            </div>
        </div>
    </div>
);


// const mapStateToProps = (state) =>  state.areas;

const mapStateToProps = (state) => ({ areas: state.areas });

export default connect(mapStateToProps)(AreaList);
