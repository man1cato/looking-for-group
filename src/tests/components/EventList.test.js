import React from 'react';
import {shallow} from 'enzyme';
import {EventList} from '../../components/EventList';
import events from '../fixtures/events';


test('should render event list with events', () => {
    const wrapper = shallow(<EventList interest={events[0].interest} area={events[0].area} events={events}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render event list with empty message', () => {
    const wrapper = shallow(<EventList events={[]}/>);
    expect(wrapper).toMatchSnapshot();
});