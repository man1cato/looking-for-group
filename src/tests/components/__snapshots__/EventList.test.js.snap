// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render event list with empty message 1`] = `
<div>
  <div
    className="list-item list-item--message"
  >
    This may take some time to load. If it has been more than a minute, then there are no available event times for this group. 
  </div>
</div>
`;

exports[`should render event list with events 1`] = `
<div>
  <Connect(EventListItem)
    area="\\"Miami-Fort Lauderdale-West Palm Beach, FL\\""
    availability="reczJFn34Yyf1bCzs"
    chatGroupUrl="https://app.groupme.com/join_group/36930410/hcggya"
    date="Dec 21st"
    groupId="recgQJVVIDDIvgUyw"
    id="rec07gX9hr16kHYgS"
    interest="Board Games"
    key="rec07gX9hr16kHYgS"
    memberCount={0}
    occurrence="Thursday Evening"
    startDatetime="2017-12-21T17:00:00"
    startTime="5:00 PM"
  />
</div>
`;
